### ISSUE TITLE
[FR]: Support the "category" item element in the article list

### ISSUE THREAD

**Issue by ceaum** [martinrotter/rssguard#746]
### Brief description of the feature request

The RSS specification includes a "category" item element . I propose adding a toggle to display this element for the items in the article list.

An example for which this may be useful is [the RSS feed of my newspaper](https://www.nrc.nl/rss/) which includes the section of an article as category element.

**Comment by martinrotter** [comment]
This feature could be useful, but at this point I do not plan to add new features, but rather stabilize existing ones and fix bugs.

The feature you ask for can be nonetheless emulated with "article filtering" feature. You can read "category" of RSS entry using `rawContents` property and prepending it (for example) to the title of entry/message.

**Comment by martinrotter** [comment]
I made an example for you which prepends RSS category to messages:
![image](https://user-images.githubusercontent.com/1255302/171570062-2a32c756-f343-4d9b-9d6f-5be02d91f36c.png)

Article filter which does this:

```js
function filterMessage() {
  // Read raw contents of message and
  // convert to JSON.
  json = utils.fromXmlToJson(msg.rawContents);
  jsonObj = JSON.parse(json);
  category = jsonObj.item.category.__text;

  msg.title = `[${category}] ${msg.title}`;

  return MessageObject.Accept;
}
```

**Comment by martinrotter** [comment]
Other way could be that RSS Guard could "generate" tags/labels for RSS categories and automagically assign those articles to relevant tags/labels, what you think?

**Comment by GeniusBroccoli** [comment]
> Other way could be that RSS Guard could "generate" tags/labels for RSS categories and automagically assign those articles to relevant tags/labels, what you think?

Yes, this is nice for big news sites, when you want filter only some categories of news.

**Comment by martinrotter** [comment]
OK. I will likely add this feature into article filtering as a "method", because it already provides a way to assign label/tag to article etc.

**Comment by cadbusca** [comment]
> I made an example for you which prepends RSS category to messages: ![image](https://user-images.githubusercontent.com/1255302/171570062-2a32c756-f343-4d9b-9d6f-5be02d91f36c.png)
> 
> Article filter which does this:
> 
> ```js
> function filterMessage() {
>   // Read raw contents of message and
>   // convert to JSON.
>   json = utils.fromXmlToJson(msg.rawContents);
>   jsonObj = JSON.parse(json);
>   category = jsonObj.item.category.__text;
> 
>   msg.title = `[${category}] ${msg.title}`;
> 
>   return MessageObject.Accept;
> }
> ```
> 
> You can, of course, do anything other you want, for example copying the category into "author" field and use that column too.
> 
> ![image](https://user-images.githubusercontent.com/1255302/171570437-1df89e18-6af5-41de-b31f-4930e9508905.png)
> 
> ```js
> function filterMessage() {
>   // Read raw contents of message and
>   // convert to JSON.
>   json = utils.fromXmlToJson(msg.rawContents);
>   jsonObj = JSON.parse(json);
>   category = jsonObj.item.category.__text;
> 
>   msg.author = category;
> 
>   return MessageObject.Accept;
> }
> ```

This workaround is not working for the case where there are more than 1 category.
Example:
For:

<category><![CDATA[jazz]]></category>
<category><![CDATA[André B. Silva]]></category>
<category><![CDATA[Clean Feed]]></category>

The Output will be only: Clean Feed
Expected: Jazz, André B. Silva, Clean Feed

**Comment by martinrotter** [comment]
@cadbusca Please, post URL of such a feed which has articles with multiple categories per article.

**Comment by bberberov** [comment]
Could you add a `category` property to the [MessageObject](https://rssguard.readthedocs.io/en/stable/features/filters.html#messageobject-class) class.  A read-only property would be just fine.  I can see how I can parse it out of `rawContents` but it would make writing filters simpler if it was already available like `author`.

Category is standardized in both [Atom](https://www.rfc-editor.org/rfc/rfc4287.html#section-4.2.2) and [RSS](https://www.rssboard.org/rss-specification#hrelementsOfLtitemgt).

**Comment by martinrotter** [comment]
It is now added to the filtering mechanism, including function to export article categories to labels. Categories from RSS, ATOM, RDF and JSON are supported.
